floater/generators.py

```diff
@@ -331,3 +333,34 @@ def subset_floats_from_mask(self, xx, yy):
         return floats_ocean
 
 
+    def npart_to_2D_array(self, var, ds1d):
+        """Constructs 2d dataset from 1d dataset of a physical variable.
+
+        PARAMETERS
+        ----------
+        var : str
+            The name of the physical variable
```

Comment:
I would prefer if the function accepted either a dataarray or a dataset. If given a dataset, it would reshape all the variables within that dataset and return a new dataset with all of those reshaped variables.